### ISSUE TITLE
[Bug]: Broken pimcore cache for relations inside objectbricks

### ISSUE THREAD

**Issue by skoch-twocream** [pimcore/pimcore#12290]
### Expected behavior

Relation in Objectbricks should work without any problems.

### Actual behavior

If the field is a mandatory field, an validation exception is thrown. If it' not a mandatory field, the relation will be removed.

It can be reproduced in the official demo:
![image](https://user-images.githubusercontent.com/54348050/171137177-ceabf5ee-95c8-4c17-9591-2b6176486bf0.png)
![image](https://user-images.githubusercontent.com/54348050/171137213-cee178b3-111a-4d34-bd79-daca2e4e8011.png)

### Steps to reproduce

1. Create a mandatory relation field inside a ojectbrick
2. Create a data object and assigne any object to relation to the new field
3. Reload the object (or open an existing object with a relation) and save it without change the relation

<hr>

## More information

The bug occurs when the cache is fetched from the cache_items table while saving the object. By opening the object and clearing the Pimcore cache before saving, everything works.

**Comment by skoch-twocream** [comment]
I think the bug just occurs since Pimcore X/10!

**Comment by fashxp** [comment]
@skochTC only reproducible in our end, when some event listener re loads the object to be saved with `force = true`. 
can you confirm that?

**Comment by skoch-twocream** [comment]
Hi @fashxp - I not sure really understand what you mean. Can you send me an example please?

**Comment by fashxp** [comment]
@skochTC does this happen all the time for you? 
for me, I only can reproduce it, when there is an event listener on the update event that re-loads the object to be saved, like so `$originalObject = AbstractObject::getById($element->getId(), true);`

**Comment by skoch-twocream** [comment]
No we dont. But we have an event listener in our project that accesses the versions:

```
        $versions = $object->getVersions();

        if (
            empty($versions)
            || (!$object instanceof DataObject\CustomerProduct && !$object instanceof DataObject\CustomerCollection)
        ) {
            return;
        }

        $lastVersion = end($versions)->getData();

        foreach (Tool::getValidLanguages() as $language) {
            if ((
                    $object instanceof DataObject\CustomerProduct
                    && $object->getProductTextLong($language) !== $lastVersion->getProductTextLong($language)
                ) || (
                    $object instanceof DataObject\CustomerCollection
                    && $object->getBody($language) !== $lastVersion->getBody($language)
                )
            ) {
                $object->setProductTextLastChange(time(), $language);
            }
        }
```

If Iremove these lines, the error disappears. Have you got a solution to check this differently (to see if a field has changed)? Like to match directly the DB value instead of the version, because we are in a preUpdate event?

What I find unclear is why I was able to reproduce this in your public demo 🤔

**Comment by kingjia90** [comment]
> What I find unclear is why I was able to reproduce this in your public demo 🤔

Confirming, had some problem in the demo, but not locally nor in the demo.pimcore.fun , but initially excluded it as a version thing because locally i have the same as the demo

**Comment by fashxp** [comment]
> What I find unclear is why I was able to reproduce this in your public demo 🤔

on demo.pimcore.com there is a bundle installed, that re loads the data object ... at least for car objects.

**Comment by skoch-twocream** [comment]
> on demo.pimcore.com there is a bundle installed, that re loads the data object ... at least for car objects.

This was the class I modified for temporary testing.

**Comment by skoch-twocream** [comment]
This issue will be kept under troubleshooting and then fixed or do we need to find a workaround for us?

Unfortunately I can contribute less at the moment.

**Comment by fashxp** [comment]
problem is, that object bricks from cached data objects don't have the reference to the object set and load them by id during runtime. and then it can happen, that it loads a different instance of the data object that the one that is actually saved ... and then it goes wrong. 

possible fix is to set the data object reference of the object brick when brick is loaded from cache (as it is done when it is loaded from database). 
we do something similar for localized fields already, see [here](https://github.com/pimcore/pimcore/blob/04c88ceb53e33e9b325dba2d5d521b9129c873ff/models/DataObject/Concrete.php#L789). for object bricks this might be a more complicated as we can have multiple brick fields per data object.

**Comment by fashxp** [comment]
same problem exists for field collections

**Comment by brusch** [comment]
Fixed by #12588
